### PR TITLE
Reworked logging to slim down code.

### PR DESCRIPTION
**Pull request by MorganVergara**
I added the common.logging framework (http://net-commons.github.io/common-logging/) to slim down the dependencies.  This allows users to log with their choice of supported logging tools (NLog, log4Net, SeriLog are all supported).

I had to move the tests to NUnit since that's our testing framework of choice.  All the tests are passing.

I also had to move to .Net 4.5.2 as we're not using .Net Core yet.  Let me know if this is a problem.  Common.Logging works with .Net Core 1.0 so you should be able to migrate it if needed.

### CODE REVIEW

File: LaunchDarklyClient/EventProcessor.cs [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+					Task.Delay(TimeSpan.FromSeconds(1)).Wait();
+					cts = new CancellationTokenSource(config.HttpClientTimeout);
+					try
+					{
+						log.Debug($"Submitting {events.Count} events to {uri.AbsoluteUri} with json: {jsonEvents}");
+						await SendEventsAsync(jsonEvents, cts);
+					}
+					catch (TaskCanceledException tce)
+					{
+						log.Error(tce.CancellationToken == cts.Token ? $"Error Submitting Events using uri: \'{uri.AbsoluteUri}\' \'{Util.ExceptionMessage(tce)}\'{tce} {tce.StackTrace}" : $"Timed out trying to send {events.Count} events after {config.HttpClientTimeout}");
```
Comment (JeffAshton) [inline LaunchDarklyClient/EventProcessor.cs]: Looses details about the exception

File: LaunchDarklyClient/EventProcessor.cs [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+						// Using a new client after errors because: https://github.com/dotnet/corefx/issues/11224
+						httpClient?.Dispose();
+						httpClient = config.HttpClient();
+					}
+					catch (Exception ex)
+					{
+						log.Error($"Error Submitting Events using uri: \'{uri.AbsoluteUri}\' \'{Util.ExceptionMessage(ex)}\'{ex} {ex.StackTrace}");
```
Comment (JeffAshton) [inline LaunchDarklyClient/EventProcessor.cs]: Looses details about the exception

**Comment by JeffAshton** [comment]
I see this PR is pretty stale, but I strongly oppose to this pull request.  It's moving logging in the wrong direction.  Log messages should never be composed from $ string semantics.

It makes it really difficult to do analysis on log messages if every message is just slightly different.  

There are good lessons to be learned from https://github.com/trentm/node-bunyan.  The message should be a constant string, and everything else should be treated as additional metadata.  For example,

```log.Error($"Error Submitting Events using uri: \'{uri.AbsoluteUri}\' \'{Util.ExceptionMessage(ex)}\'{ex} {ex.StackTrace}");```

should be broken down into

Message:  ```"Error Submitting Events"```
Uri:  ```uri.AbsoluteUri```
Exception:  ```ex```

**Comment by atrakh** [comment]
Hi @MorganVergara,

Happy to share that we've just released v4.0.0 of the .NET SDK, which migrates existing logs to the Common.Logging framework. This helps minimize dependencies brought in when targeting `net452`.

I'm going to close this PR, but please do reach out and let us know if you have any questions.

**Pull request closed without merging**